discovery-infra/test_infra/assisted_service_api.py

```diff
@@ -168,33 +178,42 @@ def generate_image(
             self,
             cluster_id: str,
             ssh_key: str,
+            pull_secret: str,
             image_type: str = consts.ImageType.FULL_ISO,
             static_network_config: Optional[list] = None,
-    ) -> models.cluster.Cluster:
+    ) -> models.infra_env.InfraEnv:
+        cluster = self.cluster_get(cluster_id=cluster_id)
         log.info("Generating image for cluster %s", cluster_id)
-        image_create_params = models.ImageCreateParams(
-            ssh_public_key=ssh_key, static_network_config=static_network_config, image_type=image_type
+        return self.create_infra_env(
```

Comment:
Are you sure this is the proper way to do it? `generate_image == create_infra_env` ?